packages/opencensus-core/src/trace/config/types.ts

```diff
@@ -59,13 +59,31 @@ export interface TracingConfig {
    * Ex.:
    * plugins: {
    *  'http': '@opencensus/opencensus-instrumentation-http',
-   *  'mongodb-core': '@opencensus/opencensus-instrumentation-mongodb-core',
+   *  'mongodb': '@opencensus/opencensus-instrumentation-mongodb',
+   *  'grpc': '@opencensus/opencensus-instrumentation-grpc',
    *   ...
    * }
    * Any user-provided value will be added to the default list.
    * It will override any default plugin for the same key.
    */
   plugins?: PluginNames;
+  /**
+   * Sometimes, there is a need to patch internal module files that are not
+   * exported by default.
+   *
+   * In those cases, those files can be set using this config property.  A name
+   * should be provided to be mapped to the file path. This path should be a
+   * relative path considering the main index file directory.
+   *
+   * Ex.: (internal files for grpc plugin that need patch)
+   *
+   *  pluginsExtra: {
+   *      'grpc':{'client':'src/client.js','metadata':'src/metadata.js'
+   *       ...
+   * }
+   *
+   */
+  pluginsExtra?: PluginExtraFiles2Patch;
```

Comment:
This is an implementation detail for each instrumentation module. I think we shouldn't add another configuration option for this reason.

Is it possible to make patching internal files a feature of the `BasePlugin`/`PluginLoader` classes?